packages/server/src/login/login-course.service.ts

```diff
@@ -154,13 +137,39 @@ export class LoginCourseService {
       '50': 'Summer_Full',
       '60': 'Summer_2',
     };
+
     // parsing time
-    let year = Number(khourySemester.slice(0, 4));
-    const season = courseSeasonMap[khourySemester.slice(-2)];
-    // edge case for Fall semester, included in the next academic year
-    if (season === 'Fall') {
-      year--;
+    let year: number;
+    let season: Season;
+    try {
+      year = Number(khourySemester.slice(0, 4));
+      season = courseSeasonMap[khourySemester.slice(-2)];
+      // edge case for Fall semester, included in the next academic year
+      if (season === 'Fall') {
+        year--;
+      }
+    } catch (e) {
+      console.log(e);
+      throw new Error(e);
```

Comment:
Yikes this is super lazy of me. Probably just gonna get rid of the try/catch later?

Reply:
I think this was completely reactionary because of an invalid test case 😆